### PR TITLE
Forward error handling to routing controller

### PR DESCRIPTION
**Pull request by aschempp**
This is a first step to get rid of our `PageErrorXXX` classes, since they are currently "hardcoded" to generate some error pages. We already have the `PrettyErrorScreenListener` that generates the page content. Instead of calling `PageError404()->getResponse()` this will not essentially call `FrontendIndex::renderPage()` with the given page type, which in turn might use the old page class. As soon as there is a real controller for an error page (possibly my next PR), this will make the page classes obsolete.

### CODE REVIEW

File: core-bundle/src/Resources/contao/models/PageModel.php [hunk 1]
```diff
@@ -531,6 +531,34 @@ public static function findFirstPublishedRegularByPid($intPid, array $arrOptions
 		return static::findOneBy($arrColumns, $intPid, $arrOptions);
 	}
 
+	/**
+	 * Find the first published regular page by its parent ID
```
Comment (Toflar) [inline core-bundle/src/Resources/contao/models/PageModel.php]: Regular?

**Review by Toflar — COMMENTED** [review]
Looks very good to me! Just a question before I approve: Don't we need the old page types for BC? We can throw them away soon ;)

File: core-bundle/src/Resources/contao/models/PageModel.php (hunk 1 above)
Comment (aschempp) [reply core-bundle/src/Resources/contao/models/PageModel.php]: fixed in dad9b92cf9e4ecdf0f44aef8ff65624953855320

**Comment by aschempp** [comment]
> Don't we need the old page types for BC?

Not exactly sure what you mean? The controller for all legacy page types is `FrontendIndex::renderPage()`, which still calls the `TL_PTY` objects: https://github.com/contao/contao/blob/4.x/core-bundle/src/Resources/contao/controllers/FrontendIndex.php#L295

**Review by Toflar — APPROVED** [review]

File: core-bundle/src/EventListener/PrettyErrorScreenListener.php [hunk 2]
```diff
@@ -132,32 +138,33 @@ private function renderErrorScreenByType(int $type, ExceptionEvent $event): void
 
         $processing = true;
 
-        if (null !== ($response = $this->getResponseFromPageHandler($type))) {
-            $event->setResponse($response);
-        }
-
-        $processing = false;
-    }
+        try {
```
Comment (leofeyer) [inline core-bundle/src/EventListener/PrettyErrorScreenListener.php:141]: The try block seems to be longer than it should be. Is it not just required around the `$this->httpKernel->handle()` method?

File: core-bundle/src/Resources/contao/models/PageModel.php [hunk 3]
```diff
@@ -531,6 +531,34 @@ public static function findFirstPublishedRegularByPid($intPid, array $arrOptions
...
+	/**
+	 * Find the first published page by its type and parent ID
+	 *
+	 * @param string  $strType    The page type
+	 * @param integer $intPid     The parent page's ID
+	 * @param array   $arrOptions An optional options array
+	 *
+	 * @return PageModel|null The model or null if there is no published regular page
+	 */
+	public static function findFirstPublishedTypeByPid($strType, $intPid, array $arrOptions=array())
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/models/PageModel.php]:
```suggestion
	public static function findFirstPublishedByTypeAndPid($strType, $intPid, array $arrOptions=array())
```

File: core-bundle/src/EventListener/PrettyErrorScreenListener.php (hunk 2 above)
Comment (aschempp) [reply core-bundle/src/EventListener/PrettyErrorScreenListener.php:141]: changed in 2752e29fc707bce414fbb0e41a3f2b0980a30c75

**Review by leofeyer — APPROVED** [review]

**Pull request merged**